### PR TITLE
Bugfix for `GAS_NEXT` of `CREATE`'s

### PR DESCRIPTION
**Pull request by OlivierBBB**
- exceptional `CREATE`'s consume all gas
- aborted `CREATE`'s consume only the upfront gas cost
- `CREATE`'s that raise the **Failure Condition F** consume the upfront gas cost + the `63/64ths` given to the child context which they don't get back
- `CREATE`'s with empty init code get those `63/64ths` back immediately so from the zkevm POV don't pay for it at all
- `CREATE`'s with nonempty init code consume the `upfront_gas_cost + 63/64ths` which they provide to the child context (and may et back in part)

### CODE REVIEW

File: hub/instruction_handling/create/generalities/all.tex [hunk 1]
```diff
@@ -234,24 +234,41 @@
...
+		\saNote{}
+		We provide some details.
+		Exceptional \inst{CREATE}-type instructions, like any exceptional instruction, fully consume the currently available gas.
+		Aborted \inst{CREATE}-type instructions only pay for the \locCreateUpfrontGasCost{} (and there is no child context.)
+		\inst{CREATE}-type instructions that raise the \textbf{Failure Condition $\bm{F}$} pay the \locCreateFullGasCost{}, including the ``\texttt{(63/64)-ths}'' to the child context, and don't get any of it back.
+		\inst{CREATE}-type instructions that lead to execution pay for \locCreateFullGasCost{}.
+		However, if the provided initialization code is \emph{empty}, then the execution context doesn't change and gets the entire ``\texttt{(63/64)-ths}'' back immediately.
+		The effective gas cost is therefore the \locCreateUpfrontGasCost{}.
+		If the provided initialization code is \emph{nonempty} then the execution context pays the \locCreateFullGasCost{}.
+		It \emph{may} get some of it back later down the line.
```
Comment (OlivierBBB) [inline hub/instruction_handling/create/generalities/all.tex:273]: This is the fix. The issue was with **Failure Condition F** raising CREATE's not following the same pattern as other scenarios in `\scenCreateNoContextChange`

Comment (OlivierBBB) [reply hub/instruction_handling/create/generalities/all.tex:273]: For context, **Failure Condition F**: here we only care about the state collision bit, i.e. the first line

image

Comment (OlivierBBB) [reply hub/instruction_handling/create/generalities/all.tex:273]: And the associated effect on returned gas
image

File: hub/instruction_handling/create/generalities/all.tex [hunk 2]
```diff
@@ -234,24 +234,41 @@
...
-						+ & \locStpGasPoop  \\
-					\end{array} \right]
-				\]
-		\end{enumerate}
+		\[
+			\hspace*{-0.5cm}
+			\left\{ \begin{array}{lcl}
+				\If \scenCreateException                 _{i} = 1 & \Then & \gasNext _{i} = 0 ~ (\trash)                               \\
+				\If \scenCreateAbort                     _{i} = 1 & \Then & \gasNext _{i} = \gasActual _{i} - \locCreateUpfrontGasCost \\
+				\If \scenCreateFCond                     _{i} = 1 & \Then & \gasNext _{i} = \gasActual _{i} - \locCreateFullGasCost    \\
```
Comment (OlivierBBB) [inline hub/instruction_handling/create/generalities/all.tex:259]: This one here is the bug fix: `scenario/CREATE_FAILURE_CONDITION` had wrong gas comuptation

**Review by lorenzogentile404 — APPROVED** [review]
LGTM

**Comment by OlivierBBB** [comment]
Was merged locally.

**Pull request closed without merging**